### ISSUE TITLE
 Why is the recombination rate calculated as 0 after adding the  --demographicHistory ？

### ISSUE THREAD

**Issue by Joyce121** [kr-colab/ReLERNN#58]
Hellow
        The reviewer expressed concerns regarding the use of the software to estimate absolute recombination rates in populations with drastically different nucleotide diversity and demographic histories. 
      While the software might handle demographic misspecification to some extent, it is limited to inferring relative recombination rates at hotspots, not absolute rates. Accurate absolute recombination maps require pedigree studies or sperm-typing. 
       Additionally, the reviewer pointed out that the software trained the neural network using an equilibrium demographic model, which does not account for the bottleneck and admixture histories of domestic breeds. 
       This leads to inaccurate estimates of absolute recombination rates, despite the authors placing significant weight on these inferred values. 
      In response, I added results from smc++, using the --demographicHistory  WL_smc++_plot.csv.   However, the recombination rate calculated for all chromosomes across all breeds is 0. 
             
     **I would appreciate your guidance on how to resolve this issue or how I should respond to the reviewer’s comments?** 
    
     
     Thank you again!       

![image](https://github.com/user-attachments/assets/d7d035ec-2cf0-4b79-8dfe-1038715ea0b3)

**Comment by andrewkern** [comment]
hi there. 

i'm going to need some more information to help you debug where things are going wrong.

does the training step look okay but not once you predict it on your data? can you plot your inferred demographic history? does that look appropriate?

**Comment by Joyce121** [comment]
How should I response the reviewer's comment? 
Could you give me some suggestions?
Thank you very much!

**Comment by andrewkern** [comment]
> Thank you for your help! I can now use the smc++ results for calculations with the --demographicHistory parameter. However, the correlation between these results and those obtained using the equilibrium demographic model is very low. Additionally, when I analyzed chromosome 6 with the same parameter (--demographicHistory), even after running it twice, the correlation between the results remained low. How can I address this issue?

Hi there-- I'm still hoping for answers to the questions I asked you above to help you debug. What does your demographic history look like? Is it appropriate?

**Comment by Joyce121** [comment]
Hello, these are my SMC++ results. Thank you!  
[GGS.pdf](https://github.com/user-attachments/files/18088067/GGS.pdf)
[WL.pdf](https://github.com/user-attachments/files/18088070/WL.pdf)

**Comment by andrewkern** [comment]
okay and is this appropriate for your organism?

If it is, the next question is how to how does the output of the training step look?

**Comment by Joyce121** [comment]
Thank you, I think it's appropriate.
Here is the  output of the training. 
[bootstrapPlot.pdf](https://github.com/user-attachments/files/18088824/bootstrapPlot.pdf)
[testResults.pdf](https://github.com/user-attachments/files/18088826/testResults.pdf)

**Comment by andrewkern** [comment]
to be clear-- is the plot called `bootstrapPlot.pdf` from the BS_CORRECT step?

the training results look very good. if you use this model on your data what do the results look like?

**Comment by Joyce121** [comment]
Yes, The plot called bootstrapPlot.pdf from the BS_CORRECT step.
And the results look like this.  
![image](https://github.com/user-attachments/assets/eea59de9-c78f-4131-8184-1be022de2921)

**Comment by andrewkern** [comment]
sorry, i don't understand what you mean by "It seems similar to using the equilibrium demographic model, but with the same data, the correlation coefficient of the results is very low."

**Comment by Joyce121** [comment]
For the same dataset, the results with and without the --demographicHistory parameter show no difference in format, but the correlation is very low. On chromosome 1, it's only around 0.5.

**Comment by andrewkern** [comment]
can you show me a scatter plot of each window estimate with and without `--demographicHistory`?

**Comment by Joyce121** [comment]
![image](https://github.com/user-attachments/assets/cd6938e1-f0eb-4722-9e02-5cf63f29f6de)

**Comment by Joyce121** [comment]
![image](https://github.com/user-attachments/assets/b79fd54f-ae0e-43b0-ac4e-918cde5ccc11)

**Comment by Joyce121** [comment]
![image](https://github.com/user-attachments/assets/1b17c35b-517b-4513-81fd-41a28b1f6f82)

**Comment by andrewkern** [comment]
so at first blush I'm not surprised by this. essentially the demographic history is creating more LD and that is reflected in the estimates of recombination rate.

**Comment by Joyce121** [comment]
Do I now have to recalculate the recombination rates with the parameter **--demographicHistory**  and replace all the results in my paper? 
Could you give me some suggestions?
Thank you very much!

**Comment by Joyce121** [comment]
Although the correlation of the results with and without the --demographicHistory parameter is not high, the main result and  conclusions of the paper remain unchanged.

**Comment by andrewkern** [comment]
i can't tell you what to do for your paper. that's up to you after considering what you are measuring and what you want to conclude.
